src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/DisableAzureDiskEncryption.cs

```diff
@@ -125,6 +125,72 @@ private Hashtable GetExtensionProtectedSettings()
             return null;
         }
 
+        private bool IsDiskEncryptionSettingPresent(VirtualMachine vm)
+        {
+            bool isPresent = false; 
+            if (vm != null && 
+                vm.InstanceView != null &&
+                vm.InstanceView.Disks != null &&
+                vm.InstanceView.Disks.Count > 0)
+            {
+                foreach ( DiskInstanceView d in vm.InstanceView.Disks )
+                {
+                    if (d.EncryptionSettings != null && d.EncryptionSettings.Count>0)
+                    {
+                        isPresent = true;
+                    }
+                }
+            }
+            return isPresent;
+        }
+
+        private string GetVersionForDisable(VirtualMachine vm)
+        {
+            // if there is currently an extension installed, use that version for disable as well 
+            if (vm.Resources != null)
+            {
+                foreach (VirtualMachineExtension vme in vm.Resources)
+                {
+                    if (vme.Publisher.Equals(AzureDiskEncryptionExtensionContext.ExtensionDefaultPublisher) &&
```

Comment:
Please remove the version check here. By doing just publisher check, you will use the existing version.